sdk/communication/AzureCommunicationCalling/ApiSamples.md

```diff
@@ -0,0 +1,758 @@
+
+## Call Basics
+<!--Start Call Basics-->
+### Call Properties
+Call object has various properties
+```swift
+// [String] caller identity
+self.call?.id
+
+// CallStateNone = 0,CallStateEarlyMedia = 1,CallStateIncoming = 2,CallStateConnecting = 3,CallStateRinging = 4,CallStateConnected = 5,CallStateHold = 6,CallStateDisconnecting = 7,CallStateDisconnected = 8
+self.call?.state
+
+// CallDirectionOutgoing = 1, CallDirectionIncoming = 2
+self.call?.callDirection
+
+// [Bool] isMicrophoneMuted - is local audio muted
+self.call?.isMicrophoneMuted
+
+// [Errror] callEndReason - containing code/subcode/message indicating how call ended
+self.call?.callEndReason
+
+// String[] localVideoStreams - collection of video streams send to other participants in a call
+self.call?.localVideoStreams
+
+// RemoteParticipant[] remoteParticipants - collection of remote participants participating in this call
+self.call?.remoteParticipants
+
+// Indicates if the call is currently being recorded or not
+self.call?.isRecordingActive;
+```
+
+### Call Operations
+* ### Make Outgoing Call
+
+To create and start a call you need to call one of the APIs on CallAgent
+```swift
+public class CallManager : NSObject, CallDelegate
+{
+    public func placeCall(names: [CommunicationIdentifier])
+    {
+        let callOptions = StartCallOptions()!
+        // Using the CallAgent received from the factory method
+        // and returns a Call object
+        let audioOptions = AudioOptions()
+        audioOptions!.muted = true // Start the call in mute state
+        callOptions.audioOptions = audioOptions
+        let call = self.callAgent?.startCall(participants: names,
+                                            options: callOptions)
+        call?.delegate = self.callObserver
+    }
+
+    public func placeVideoCall(names: [CommunicationIdentifier])
+    {
+        let camera = self.deviceManager!.cameras!.first
+        let localVideoStream = LocalVideoStream(camera: camera)
+        let videoOptions = VideoOptions(localVideoStream: localVideoStream)
+        
+        let callOptions = StartCallOptions()
+        callOptions?.videoOptions = videoOptions
+
+        let call = self.callAgent?.startCall(participants: names,
+                                                options: callOptions)
+        call!.delegate = self.callObserver
+    }
+
+    // Following is one event which is part of CallDelegate
+    public func OnStateChanged(_ call: Call!,
+                                args: PropertyChangedEventArgs!)
+    {
+        let state = CallObserver.callStateToString(state: call.state)
+        print("Call state changed to %@", state)
+    }
+}
+
+```
+
+* ### Accept Incoming Call
+
+To accept a call you need to call one of the APIs on CallAgent
+```swift
+public class CallManager : NSObject
+{
+    public func acceptCall()
+    {
+        let acceptCallOptions = AcceptCallOptions()!
+        let camera = self.deviceManager?.cameras!.first
```

Comment:
Does this make sense for `cameras` to be nullable? I would expect that if the device manager didn't have any cameras to just return an empty array, or throw

Reply:
cameras will never be null or empty array but app developer should still take care of edge cases when say  there is an operating system bug and cameras are not listed for some weird reason. 
```
        guard let camera = deviceManager.getCameraList()!.first else {
            print("")
            return
        }

        localVideoStream = LocalVideoStream(camera: camera)
        completionHandler(localVideoStream)
```
I havent added gaurd statements all over the sample to keep the length of the sample to minimum and this code will not be published to the users.

Reply:
I suspect this is a result of the ObjC translation where pretty much anything can be nil.

Reply:
Unfortunate. Should be preface them on the ObjC side with Non_Null?

Reply:
Let me correct this, i investigated on this and ApiGen support for setting nullable type specifier will fix this issue. As mentioned previously this is WIP and will be fixed before GA.